tensorpack/cmtf.py

```diff
@@ -226,12 +226,15 @@ def perform_CP(tOrig, r=6, tol=1e-6, maxiter=50, progress=False):
 
     R2X_last = -np.inf
     tFac.R2X = calcR2X(tFac, tOrig)
+    if callback:
+        metric = 0
```

Comment:
I originally left this in case we wanted to switch it to runtime, but I wasn't sure how it might be implemented (`time` library?). I assume tFac object cannot track the runtime on its own, so the tracker creates an (n x 2) matrix that can hold iteration or runtime (this is also why I have the counter attribute in the tracker object).

Reply:
Good thinking, but the callback doesn't need this to track time. It can simply record when it was first called (`time.time()` is sufficiently accurate), and then look at the difference in subsequent calls.

If the tracker simply appends to an array (`np.append()`), then it doesn't need to store the iteration number anywhere. You always will know `x[0]` is iteration 1, `x[1]` is iteration 2, etc, and the length is the total number of iterations. It's not the most efficient way of doing things, but the efficiency loss won't matter here, and it will be more readable.